compiler/noirc_frontend/src/elaborator/statements.rs

```diff
@@ -206,9 +206,13 @@ impl<'context> Elaborator<'context> {
     }
 
     fn elaborate_jump(&mut self, is_break: bool, span: noirc_errors::Span) -> (HirStatement, Type) {
-        let in_constrained_function = self
-            .current_function
-            .map_or(true, |func_id| !self.interner.function_modifiers(&func_id).is_unconstrained);
+        let in_constrained_function = self.current_item.map_or(true, |id| match id {
+            DependencyId::Function(func_id) => {
+                !self.interner.function_modifiers(&func_id).is_unconstrained
+            }
+            _ => true,
+        });
```

Comment:
```suggestion
        let in_constrained_function = self.current_item.map_or(true, |id| match id {
            DependencyId::Function(id) => !self.interner.function_modifiers(&id).is_unconstrained,
            _ => true,
        });
```

Copying some changes from below to get reduce number of lines.

Reply:
We could also move this out into a helper function similar to `in_comptime_context`

Reply:
I ended up adding a helper function